### PR TITLE
[2] Add route for remove a pipeline

### PR DESCRIPTION
**Pull request by d2lam**
- Adding a route for removing a pipeline 
- Refactor some code in test
- Blocked by https://github.com/screwdriver-cd/models/pull/109
- Relates to https://github.com/screwdriver-cd/models/issues/105

### CODE REVIEW

File: plugins/pipelines/remove.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+const schema = require('screwdriver-data-schema');
+const idSchema = joi.reach(schema.models.pipeline.base, 'id');
+
+module.exports = () => ({
+    method: 'DELETE',
+    path: '/pipelines/{id}',
+    config: {
+        description: 'Delete a single pipeline',
+        notes: 'Returns null if successful',
+        tags: ['api', 'pipelines'],
```
Comment (stjohnjohnson) [inline plugins/pipelines/remove.js]: I think you need to add an auth check here.

File: plugins/pipelines/remove.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        handler: (request, reply) => {
+            const factory = request.server.app.pipelineFactory;
+
+            return factory.get(request.params.id)
+                .then(pipeline => {
+                    if (!pipeline) {
+                        throw boom.notFound('Pipeline does not exist');
+                    }
+
+                    return reply(pipeline.remove());
```
Comment (stjohnjohnson) [inline plugins/pipelines/remove.js]: So anyone can remove pipelines?  That's cool.

File: plugins/pipelines/remove.js (hunk 1 above)
Comment (d2lam) [inline plugins/pipelines/remove.js]: good call!

Comment (d2lam) [inline plugins/pipelines/remove.js]: @stjohnjohnson  Addressed

**Pull request merged**